frontend/src/components/playground/SaveStatusIndicator.tsx

```diff
@@ -0,0 +1,74 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import { SaveStatus, SaveState } from "@/types/SaveStatus";
+import StatusIndicator from "@cloudscape-design/components/status-indicator";
+
+import { formatDistanceToNow } from "date-fns";
+
+/**
+ * Component to display the save status of a document or transformation
+ */
+interface SaveStatusIndicatorProps {
+  state: SaveState;
+}
+
+/**
+ * A component that displays a status indicator for the save state of a document or transformation
+ *
+ * @param status - The current save status (SAVED, UNSAVED, or BLOCKED)
+ */
+export default function SaveStatusIndicator({
+  state,
+}: Readonly<SaveStatusIndicatorProps>) {
+  // Add a state to force re-renders of the "X seconds ago" text
+  const [, setTick] = useState(0);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setTick((tick) => tick + 1);
+    }, 1000 * 30); // 30 seconds
+
+    // Clean up on unmount
+    return () => clearInterval(intervalId);
+  }, []);
+  const getStatusType = (state: SaveState) => {
+    switch (state?.status ?? null) {
+      case SaveStatus.SAVED:
+        return "success";
+      case SaveStatus.BLOCKED:
+        return "error";
+      case SaveStatus.UNSAVED:
+      default:
+        return "in-progress";
+    }
+  };
+
+  const getStatusText = (state: SaveState) => {
+    switch (state?.status ?? null) {
+      case SaveStatus.SAVED:
+        return (
+          "Saved" +
+          (state.savedAt ? ` ${formatDistanceToNow(state.savedAt)} ago` : "")
+        );
+      case SaveStatus.BLOCKED:
+        return (
+          "Blocked" +
+          (state.errors.length
+            ? ` (${state.errors.length} error${
+                state.errors.length > 1 ? "s" : ""
```

Comment:
I'm not sure I agree with sonarqube on this one, I think it could be abused, but this use is fine.  I'd be ok if you wanted to add an except for this rule on all .tsx files

Reply:
Already fixed it, but I don't disagree with you. I was considering it for this and the other one in the file.